### PR TITLE
[#5] Implement UI My coins list on home page

### LINKED ISSUE #5: [UI] As a user, I can see My Coins list on Home page
## Why

When the user launches the app, it will show the "My Coins" section on the Home page.

## Acceptance Criteria

- Show the following UI:
  - Header: "My Coins 😎"
  - "see all" with arrow icon
  - List of coins on horizontal scrolling
- Coin item:
  - Logo
  - Coin symbol (BTC, ETH, etc.)
  - Coin name (Bitcoin, Ethereum, etc.)
  - Price
  - Price change percentage in 24h
    - Positive number: Use green color with arrow up `#10DC78`
    - Negative number: Use red color with arrow down `#F15950`
 - Display adaptive in light/dark mode

## Design

image

## Resources

https://www.figma.com/file/HioYp71LllKVy9qaiLauKn/Crypto-App-(Community)?node-id=3%3A244

### PR DESCRIPTION
**Pull request by Wadeewee**
#5

## What happened 👀

When the user launches the app, it will show the "My Coins" section on the Home page.

- Show the following UI:
  - Header: "My Coins 😎"
  - "see all" with arrow icon
  - List of coins on horizontal scrolling
- Coin item:
  - Logo
  - Coin symbol (BTC, ETH, etc.)
  - Coin name (Bitcoin, Ethereum, etc.)
  - Price
  - Price change percentage in 24h
     - Positive number: Use green color with arrow up #10DC78
     - Negative number: Use red color with arrow down #F15950
- Display adaptive in light/dark mode

## Insight 📝

- Create `CoinItem` and `SeeAll` Component.
- Update Style and Theme.

**Note:** We will remove `Typography.kt` file in this PR as we don't use the material theme for font style and update `font family` in `TextStyle` instead. Please have a look in the `Style` class.

## Proof Of Work 📹

https://user-images.githubusercontent.com/28002315/180743362-2d0d5687-a663-4547-8b29-418c6b3044ca.mov

### CODE REVIEW

**Comment by luongvo** [comment]
@Wadeewee Could you please rebase to #11 then I can have a review? 👋

**Comment by luongvo** [comment]
@Wadeewee Should we drop these commits after rebasing?

image

**Comment by hoangnguyen92dn** [comment]
@Wadeewee Please update to get rid of issue reported by Detekt 🙏

**Review by hoangnguyen92dn — APPROVED** [review]
LGTM 🚀

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt [hunk 1]
```diff
@@ -37,18 +47,55 @@ fun HomeScreen() {
...
+                modifier = Modifier
+                    .clickable(onClick = { /* TODO: Update on Integrate ticket */ })
+                    .constrainAs(seeAll) {
+                        top.linkTo(myCoinsTitle.top)
+                        end.linkTo(parent.end)
+                        width = Dimension.preferredWrapContent
+                    }
+            )
+
+            LazyRow(
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt:83]: I think we need to add content padding here. The list items should be scrolled horizontally through the screen edges.

```suggestion
            LazyRow(
                contentPadding = PaddingValues(horizontal = Dp16),
```

https://user-images.githubusercontent.com/16315358/184799641-0118f4db-f039-47e0-b680-bbb39a752473.mp4

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt [hunk 2]
```diff
@@ -37,18 +47,55 @@ fun HomeScreen() {
...
+
+            SeeAll(
+                modifier = Modifier
+                    .clickable(onClick = { /* TODO: Update on Integrate ticket */ })
+                    .constrainAs(seeAll) {
+                        top.linkTo(myCoinsTitle.top)
+                        end.linkTo(parent.end)
+                        width = Dimension.preferredWrapContent
+                    }
+            )
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt]:
```suggestion
            SeeAll(
                modifier = Modifier
                    .clickable(onClick = { /* TODO: Update on Integrate ticket */ })
                    .constrainAs(seeAll) {
                        top.linkTo(myCoinsTitle.top)
                        end.linkTo(parent.end)
                        width = Dimension.preferredWrapContent
                    }
                    .padding(horizontal = Dp16)
            )
```

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt [hunk 3]
```diff
@@ -37,18 +47,55 @@ fun HomeScreen() {
...
+                modifier = Modifier.constrainAs(myCoinsTitle) {
+                    top.linkTo(portfolioCard.bottom, margin = Dp52)
+                    start.linkTo(parent.start)
+                    width = Dimension.preferredWrapContent
+
+                },
+                text = stringResource(id = R.string.home_my_coins_title),
+                style = Style.medium16(),
+                color = MaterialTheme.colors.textColor
+            )
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt]:
```suggestion
            Text(
                modifier = Modifier
                    .constrainAs(myCoinsTitle) {
                        top.linkTo(portfolioCard.bottom, margin = Dp52)
                        start.linkTo(parent.start)
                        width = Dimension.preferredWrapContent

                    }
                    .padding(horizontal = Dp16),
                text = stringResource(id = R.string.home_my_coins_title),
                style = Style.medium16(),
                color = MaterialTheme.colors.textColor
            )
```

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt [hunk 4]
```diff
@@ -24,7 +28,13 @@ fun HomeScreen() {
                 .fillMaxSize()
                 .padding(horizontal = Dp16)
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt]: Then we need to remove the padding from the root `ConstraintLayout` and add them into children composes instead 🤔 
```suggestion
```

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt [hunk 5]
```diff
@@ -37,18 +47,55 @@ fun HomeScreen() {
...
                 style = Style.semiBold24(),
-                color = MaterialTheme.colors.titleHome
+                color = MaterialTheme.colors.textColor
             )
 
             PortfolioCard(
                 modifier = Modifier.constrainAs(portfolioCard) {
                     top.linkTo(title.bottom, margin = Dp40)
                 }
             )
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt]:
```suggestion
            PortfolioCard(
                modifier = Modifier
                    .constrainAs(portfolioCard) {
                        top.linkTo(title.bottom, margin = Dp40)
                    }
                    .padding(horizontal = Dp16)
            )
```

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/CoinItem.kt [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+@Suppress("FunctionNaming", "LongMethod")
+@Composable
+fun CoinItem(
+    isPositiveNumber: Boolean = false /* TODO Update value to Object on Integrate ticket */
+) {
+    ConstraintLayout(
+        modifier = Modifier
+            .wrapContentWidth()
+            .clip(RoundedCornerShape(Dp12))
+            .background(color = MaterialTheme.colors.coinItemColor)
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/CoinItem.kt:46]: Why don't we set the padding 8 to root layout instead?

```suggestion
    ConstraintLayout(
        modifier = Modifier
            .wrapContentWidth()
            .clip(RoundedCornerShape(Dp12))
            .background(color = MaterialTheme.colors.coinItemColor)
            .padding(horizontal = Dp8, vertical = Dp8)
```

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/CoinItem.kt [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+            coinName,
+            price,
+            icon,
+            priceChange
+        ) = createRefs()
+
+        Image(
+            modifier = Modifier
+                .size(Dp60)
+                .padding(start = Dp8, end = Dp16, top = Dp8)
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/CoinItem.kt]: then we can remove these manual padding on each item, e.g.

```suggestion
                .padding(end = Dp16)
```

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/PortfolioCard.kt [hunk 8]
```diff
@@ -27,6 +27,7 @@ import co.nimblehq.compose.crypto.ui.theme.Dimension.Dp40
 import co.nimblehq.compose.crypto.ui.theme.Dimension.Dp8
 import co.nimblehq.compose.crypto.ui.theme.Style
 
+@Suppress("FunctionNaming", "LongMethod")
 @Composable
 fun PortfolioCard(
```
Comment (luongvo) [inline app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/PortfolioCard.kt:32]: same on this card

```
    ConstraintLayout(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(Dp12))
            .background(
                brush = Brush.linearGradient(
                    colors = listOf(Color.MetallicSeaweed, Color.TiffanyBlue),
                )
            )
            .padding(horizontal = Dp16, vertical = Dp16)
```

**Review by luongvo — COMMENTED** [review]
rest lgtm

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/CoinItem.kt (hunk 7 above)
Comment (Wadeewee) [reply app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/CoinItem.kt]: @luongvo Good catch 👍 I will update 🚀

File: app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt (hunk 4 above)
Comment (Wadeewee) [reply app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt]: Hmm 🤔 @luongvo If we remove `padding` in the root, so we need to put it in each `child` instead, but I don't think it's a good idea as we need to set the `padding` in many places 🙏

Comment (Wadeewee) [reply app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt]: Oh, I see why you suggested removing `padding` in the root so we can use `contentPadding` in `LazyRow` regarding this [comment](https://github.com/nimblehq/jetpack-compose-crypto/pull/12#discussion_r946318903) 👍  Good point @luongvo

Comment (luongvo) [reply app/src/main/java/co/nimblehq/compose/crypto/ui/screens/home/HomeScreen.kt]: @Wadeewee Exactly 👍  Actually we can group the above composes into a group with parent margin but may be not enough for this time 🤓

**Review by luongvo — APPROVED** [review]
LGTM

**Review by hoangnguyen92dn — APPROVED** [review]

**Pull request merged**